DllImportGenerator/DllImportGenerator/Marshalling/CustomNativeTypeMarshaler.cs

```diff
@@ -0,0 +1,240 @@
+            if (!info.IsManagedReturnPosition &&
+                !info.IsByRef &&
+                context.PinningSupported &&
+                (_marshallingMethods & SupportedMarshallingMethods.Pinning) != 0)
+            {
+                if (context.CurrentStage == StubCodeContext.Stage.Pin)
+                {
+                    yield return FixedStatement(
+                        VariableDeclaration(
+                            PointerType(PredefinedType(Token(SyntaxKind.VoidKeyword))),
+                            SingletonSeparatedList(
+                                VariableDeclarator(Identifier(nativeIdentifier))
+                                    .WithInitializer(EqualsValueClause(
+                                        IdentifierName(managedIdentifier)
+                                    ))
+                            )
+                        ),
+                        EmptyStatement()
+                    );
+                }
+                yield break;
+            }
+            switch (context.CurrentStage)
+            {
+                case StubCodeContext.Stage.Setup:
+                    if (!_valuePropertyRequiresPinning)
+                    {
+                        yield return LocalDeclarationStatement(
+                        VariableDeclaration(
+                            _nativeTypeSyntax,
+                            SingletonSeparatedList(
+                                VariableDeclarator(nativeIdentifier))));
+                    }
+                    
+                    if (_useValueProperty)
+                    {
+                        yield return LocalDeclarationStatement(
+                            VariableDeclaration(
+                                _nativeLocalTypeSyntax,
+                                SingletonSeparatedList(
+                                    VariableDeclarator(marshalerIdentifier)
+                                        .WithInitializer(EqualsValueClause(LiteralExpression(SyntaxKind.DefaultLiteralExpression))))));
+                    }
+                    break;
+                case StubCodeContext.Stage.Marshal:
+                    if (!info.IsManagedReturnPosition && info.RefKind != RefKind.Out)
+                    {
+                        bool scenarioSupportsStackalloc = context.StackSpaceUsable &&
+                            (_marshallingMethods & SupportedMarshallingMethods.ManagedToNativeStackalloc) != 0;
+
+                        List<ArgumentSyntax> arguments = new List<ArgumentSyntax>
+                        {
+                            Argument(IdentifierName(managedIdentifier))
+                        };
+
+                        if (scenarioSupportsStackalloc && (!info.IsByRef || info.RefKind == RefKind.In))
+                        {
+                            // stackalloc byte[<_nativeLocalType>.StackBufferSize]
```

Comment:
We should set the no-zero init flag on the method - if it is not set globally on the assembly already.

Reply:
@jkotas Is the intent here avoid the cost of zero initialing the stack space?

Reply:
Yup